### DISCUSSION TITLE
Implement Run or Raise

### DISCUSSION THREAD

**Discussion by wesbluemarine** [YaLTeR/niri#46] [Ideas]
It could be functional to "teleport" to an already running app.
[Here](https://github.com/labwc/labwc/issues/673) an example.

**Comment by YaLTeR** [comment]
Once xdg_activation is implemented (#30), this should work out of the box for single-instance apps that support it.

**Reply by Schweber** [reply]
Will this entail the ability of being able to run, show *and hide* a program with a keybind (or in niri's case maybe "launch the program, if it's already running then teleport the program to the end of the strip or back into view")? 

My workflow is heavily dependent on such a dropdown-like behaviour that e.g. [tdrop](https://github.com/noctuid/tdrop) makes available in X and that i emulated for `Hyprland` in [hdrop](https://github.com/Schweber/hdrop).

**Reply by YaLTeR** [reply]
> launch the program, if it's already running then teleport the program to the end of the strip or back into view

This will be covered by IPC, see also https://github.com/YaLTeR/niri/discussions/267
